redash/authentication/__init__.py

```diff
@@ -43,15 +43,14 @@ def sign(key, path, expires):
 def load_user(user_id_with_identity):
     org = current_org._get_current_object()
 
-    user_id, _ = user_id_with_identity.split("-")
-
     try:
+        user_id, _ = user_id_with_identity.split("-")
         user = models.User.get_by_id_and_org(user_id, org)
         if user.is_disabled or user.get_id() != user_id_with_identity:
             return None
 
         return user
-    except models.NoResultFound:
+    except (models.NoResultFound, Exception):
```

Comment:
Maybe we should only catch `SyntaxError` instead of `Exception` here?

Reply:
Well it could be either a `ValueError` when `user_id_with_identity` isn't hyphenated or a `AttributeError` when `user_id_with_identity` is an integer, so I thought I'd just catch all.

Reply:
My concern here is that it might swallow some unrelated exceptions and make debugging future issues harder. Aside from this there is not much harm in catching `Exception` here.